src/test/java/EOorg/EOeolang/EOthreads/EOmutexTest.java

```diff
@@ -0,0 +1,165 @@
+            );
+            ++counter;
+        }
+        MatcherAssert.assertThat(
+            (int) counter,
+            Matchers.equalTo(hashcodes.size())
+        );
+    }
+
+    // @todo #33:90min Uncomment this test when problem
+    //  of the of same hashcodes in parallel threads
+    //  will be solved. This test proves new Phi objects
+    //  created in parallel threads can have repeated
+    //  hashcodes
+    //@Test
+    //public void PhiUniqueHashesInDynamic() throws InterruptedException {
+    //    int threads = 8;
+    //    Set<Integer> hashes = ConcurrentHashMap.newKeySet();
+    //    ExecutorService service = Executors.newFixedThreadPool(threads);
+    //    CountDownLatch latch = new CountDownLatch(1);
+    //    for (long t = 0; t < threads; ++t) {
+    //        final long finalT = t;
+    //        service.submit(
+    //            () -> {
+    //                latch.await();
+    //                Phi number = new PhWith(
+    //                    new EOnumber(Phi.Φ),
+    //                    "n",
+    //                    new Data.ToPhi(finalT)
+    //                );
+    //                System.out.println("Hashcode = " + number.hashCode() + ", n = " + finalT);
+    //                if (!hashes.add(number.hashCode())){
+    //                    System.out.println("Repeated hashcode = " + number.hashCode());
+    //                }
+    //                return number.hashCode();
+    //            }
+    //        );
+    //    }
+    //    latch.countDown();
+    //    service.awaitTermination(1, TimeUnit.SECONDS);
+    //    MatcherAssert.assertThat(
+    //        hashes.size(),
+    //        Matchers.equalTo(threads)
+    //    );
+    //}
+
+    @Test
+    public void acquisitionUpdateDecrease() throws InterruptedException {
+        final int threads = 10;
+        final ExecutorService service = Executors.newFixedThreadPool(threads);
+        final Set<Phi> phis = new HashSet<>();
+        for (long counter = 0; counter < 1e2; ++counter) {
+            phis.add(
+                new EOmutex$EOacquire(Phi.Φ)
+            );
+        }
+        final CountDownLatch latch = new CountDownLatch(1);
+        for (final Phi item: phis) {
+            service.submit(
+                () -> {
+                    latch.await();
+                    Acquisitions.INSTANCE.update(item, 4);
+                    Acquisitions.INSTANCE.decrease(item, 2);
+                    Acquisitions.INSTANCE.decrease(item, 2);
+                    return true;
+                }
+            );
+        }
+        latch.countDown();
+        service.shutdown();
+        MatcherAssert.assertThat(
+            service.awaitTermination(1, TimeUnit.SECONDS),
+            Matchers.equalTo(true)
+        );
+    }
+
+    // @todo #33:90min Implement test like the test
+    //  before where new Phi objects are created in
+    //  parallel instead of one thread. We need to
+    //  wait for the solving of hashcode problem
+
```

Comment:
@levBagryansky It is better to avoid empty lines between comments and code. Maybe we can move this `todo` before `acquisitionUpdateDecrease()` test?

Reply:
@Graur I wanted to arrange it here because it is improvement of `acquisitionUpdateDecrease()` test

Reply:
@levBagryansky If this is improvement why you want to arrange it? Sounds unclear for me